src/libAtomVM/bitstring.c

```diff
@@ -119,10 +140,64 @@ bool bitstring_utf8_encode(avm_int_t c, uint8_t *buf, size_t *out_size)
     return true;
 }
 
+bool bitstring_utf8_decode(int32_t *c, uint8_t *buf, size_t len, size_t *out_size)
+{
+    int32_t v = 0;
```

Comment:
as a personal preference, I prefer declaring a variable outside the if block only if we are going to use later.